src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3405,6 +3408,45 @@ void ApplicationManagerImpl::OnHMILevelChanged(
     mobile_apis::HMILevel::eType from,
     mobile_apis::HMILevel::eType to) {
   LOG4CXX_AUTO_TRACE(logger_);
+  LOG4CXX_INFO(logger_, "HMI level from = " << from << " to = " << to);
+  ProcessPostponedMessages(app_id);
+  ProcessNaviApp(app_id, from, to);
```

Comment:
@LitvinenkoIra it is not clear that you call ProcessNaviApp for each application, not only navi